### ISSUE TITLE
Optimize PlayServerInterpreter.toRoutes

### ISSUE THREAD

**Issue by kciesielski** [softwaremill/tapir#3549]
The method takes a lot of CPU when running the `SimpleGetMultiRoute` simulation:

![image](https://github.com/softwaremill/tapir/assets/1413553/9b98be74-9184-4ea8-af70-f8051c822851)

This affects throughput dramatically:

![image](https://github.com/softwaremill/tapir/assets/1413553/47ba001b-a445-4090-bb45-cad5289eeb63)

**Comment by dmtran-g** [comment]
Hi @kciesielski 

I would like to help on improving the `SimpleGetMultiRoute` scenario for Play. 
Any tips / insights on where to start?

I ran `perfTests/Gatling/testOnly sttp.tapir.perf.SimpleGetMultiRouteSimulation` 3 times with 128 users, and I see numbers similar to the ones shared in this issue description:
- play.VanillaMulti - mean requests/sec: 12263, 13418, 13269
- play.TapirMulti - mean requests/sec: 1131, 1190, 1044


Note: the change in https://github.com/softwaremill/tapir/pull/3838 doesn't improve the `SimpleGetMultiRoute` scenario, the numbers are similar.

**Comment by kciesielski** [comment]
Hi @dmtran-g, thanks a lot for taking a shot at this! Optimizing that interpreter instantiation in #3838 is definitely worth merging. 
Another thing I found today is how the performance test setup is done. The profiler shows that CPU is overwhelmed by calls to `PlayServerInterpreter.toRoutes`, and it raised my suspicion, because `toRoutes` seems like a method that's supposed to be called only to initialize the server.
Then, I found this in `sttp.tapir.perf.play.Play`:

![image](https://github.com/softwaremill/tapir/assets/1413553/5f0e3e22-4770-4151-9f6d-9f337b0b8d21)

This def looks suspicious, so I changed it to `lazy val` and voila, throughput for SimpleGet reached 160k reqs/sec! I'll do some more tests and profiling, but I'm pretty sure this will be the solution - to acknowledge that `router` is referenced for each request and make it return a `Router` that's initialized only once.
